src/dmd/dtemplate.d

```diff
@@ -1338,6 +1338,11 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
         // Set up scope for parameters
         Scope* paramscope = scopeForTemplateParameters(ti,sc);
 
+        // Mark the parameter scope as deprecated if the templated
+        // function is deprecated (since paramscope.enclosing is the
+        // calling scope already)
+        paramscope.stc |= fd.storage_class & STC.deprecated_;
```

Comment:
Wouldn't `scopeForTemplateParameters` be a better fit for this ??

Reply:
Maybe but then fd needs to be passed to scopeForTemplateParameters as well, hence resulting in a larger  diff

Reply:
Nevermind, `scopeForTemplateParameters` is also used inside of `matchWithInstance` wich does not handle a `FuncDeclaration`.